### PR TITLE
CU/7073-7074-Dylan-SMUnitTests

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
All SM Unit tests

## Screenshots/Video
N/A

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
M src/screens/HealthScreen/SecureMessaging/Inbox/Inbox.test.tsx
M src/screens/HealthScreen/SecureMessaging/CancelConfirmations/ComposeCancelConfirmation.test.tsx
S src/screens/HealthScreen/SecureMessaging/NoInboxMessages/NoInboxMessages.test.tsx
S src/screens/HealthScreen/SecureMessaging/ReplyMessageButton/ReplyMessageButton.test.tsx
S src/screens/HealthScreen/SecureMessaging/TermsAndConditions/TermsAndConditions.test.tsx
S src/screens/HealthScreen/SecureMessaging/Folders/Folders.test.tsx
S src/screens/HealthScreen/SecureMessaging/ReplyHelp/ReplyHelp.test.tsx
S src/screens/HealthScreen/SecureMessaging/ViewMessage/IndividualMessageErrorComponent.test.tsx
L src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/HealthScreen/SecureMessaging/CancelConfirmations/ComposeCancelConfirmation.test.tsx [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import 'react-native'
 import React from 'react'
-// Note: test renderer must be required after react-native.
-import 'jest-styled-components'
-import { ReactTestInstance, act } from 'react-test-renderer'
 
-import { context, mockNavProps, render, RenderAPI } from 'testUtils'
+import { act } from 'react-test-renderer'
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/CancelConfirmations/ComposeCancelConfirmation.test.tsx:4]: I don't think we need to use `act`in any of these tests

**Review by theodur — CHANGES_REQUESTED** [review]
Seeing issues as [here](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7148#pullrequestreview-1706588458)

**Comment by Sparowhawk** [comment]
> Seeing issues as [here](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7148#pullrequestreview-1706588458)

updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/CancelConfirmations/ComposeCancelConfirmation.test.tsx (hunk 1 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/CancelConfirmations/ComposeCancelConfirmation.test.tsx:3]: this is needed, fireEvent.press wasn't working cause it used useDestructiveActionSheet hook

Comment (theodur) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/CancelConfirmations/ComposeCancelConfirmation.test.tsx:3]: I was curious, so I tried removing them locally, and the tests still passed for me. Can you confirm on your end also?

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx [hunk 2]
```diff
@@ -110,13 +93,7 @@ context('Attachments', () => {
...
         jest.spyOn(DocumentPicker, 'pickSingle').mockReturnValue(promise)
 
-        const buttons = testInstance.findAllByType(VAButton)
-        expect(buttons.length).toEqual(1)
-        expect(buttons[0].props.label).toEqual('Select a file')
-
-        await waitFor(() => {
-          buttons[0].props.onPress()
-        })
+        fireEvent.press(screen.getByText('Select a file'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:96]:
```suggestion
        fireEvent.press(screen.getByRole('button', { name: 'Select a file' }))
```

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx [hunk 3]
```diff
@@ -127,8 +104,7 @@ context('Attachments', () => {
         await waitFor(() => {
           promise
         })
-
-        testInstance.findAllByType(VAButton)[0].props.onPress()
+        fireEvent.press(screen.getByText('Attach'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:107]:
```suggestion
        fireEvent.press(screen.getByRole('button', { name: 'Attach' }))
```

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx [hunk 4]
```diff
@@ -138,12 +114,7 @@ context('Attachments', () => {
         const failCasePromise = Promise.resolve({ uri: 'uri', name: 'custom-file-name.docx', type: 'docx', size: 90000000 } as DocumentPickerResponse)
         jest.spyOn(DocumentPicker, 'pickSingle').mockReturnValue(failCasePromise)
 
-        const allButtons = testInstance.findAllByType(VAButton)
-        expect(allButtons[0].props.label).toEqual('Select a file')
-
-        await waitFor(() => {
-          allButtons[0].props.onPress()
-        })
+        fireEvent.press(screen.getByText('Select a file'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:117]:
```suggestion
        fireEvent.press(screen.getByRole('button', { name: 'Select a file' }))
```

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx [hunk 5]
```diff
@@ -155,20 +126,15 @@ context('Attachments', () => {
...
           const failCasePromise = Promise.resolve({ uri: 'uri', name: 'custom-file-name.docx', type: 'error' } as DocumentPickerResponse)
           jest.spyOn(DocumentPicker, 'pickSingle').mockReturnValue(failCasePromise)
 
-          const allButtons = testInstance.findAllByType(VAButton)
-          expect(allButtons[0].props.label).toEqual('Select a file')
-
-          await waitFor(() => {
-            allButtons[0].props.onPress()
-          })
+          fireEvent.press(screen.getByText('Select a file'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:137]:
```suggestion
          fireEvent.press(screen.getByRole('button', { name: 'Select a file' }))
```

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx [hunk 6]
```diff
@@ -189,12 +155,7 @@ context('Attachments', () => {
           const failCasePromise = Promise.resolve({ uri: 'uri', name: 'custom-file-name.docx', type: 'docx', size: 90000000 } as DocumentPickerResponse)
           jest.spyOn(DocumentPicker, 'pickSingle').mockReturnValue(failCasePromise)
 
-          const allButtons = testInstance.findAllByType(VAButton)
-          expect(allButtons[0].props.label).toEqual('Select a file')
-
-          await waitFor(() => {
-            allButtons[0].props.onPress()
-          })
+          fireEvent.press(screen.getByText('Select a file'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:158]:
```suggestion
          fireEvent.press(screen.getByRole('button', { name: 'Select a file' }))
```

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx [hunk 7]
```diff
@@ -217,12 +178,7 @@ context('Attachments', () => {
           const failCasePromise = Promise.resolve({ uri: 'uri', name: 'custom-file-name.docx', type: 'docx', size: 1000 } as DocumentPickerResponse)
           jest.spyOn(DocumentPicker, 'pickSingle').mockReturnValue(failCasePromise)
 
-          const allButtons = testInstance.findAllByType(VAButton)
-          expect(allButtons[0].props.label).toEqual('Select a file')
-
-          await waitFor(() => {
-            allButtons[0].props.onPress()
-          })
+          fireEvent.press(screen.getByText('Select a file'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:181]:
```suggestion
          fireEvent.press(screen.getByRole('button', { name: 'Select a file' }))
```

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx [hunk 8]
```diff
@@ -245,12 +201,7 @@ context('Attachments', () => {
           const failCasePromise = Promise.resolve({ uri: 'uri1', name: 'custom-file-name.docx', type: 'docx', size: 1000 } as DocumentPickerResponse)
           jest.spyOn(DocumentPicker, 'pickSingle').mockReturnValue(failCasePromise)
 
-          const allButtons = testInstance.findAllByType(VAButton)
-          expect(allButtons[0].props.label).toEqual('Select a file')
-
-          await waitFor(() => {
-            allButtons[0].props.onPress()
-          })
+          fireEvent.press(screen.getByText('Select a file'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:204]:
```suggestion
          fireEvent.press(screen.getByRole('button', { name: 'Select a file' }))
```

File: VAMobile/src/screens/HealthScreen/SecureMessaging/ViewMessage/MessageCard.test.tsx [hunk 9]
```diff
@@ -37,65 +35,60 @@ jest.mock('store/slices', () => {
...
-      expect(downloadFileAttachment).toBeCalledWith(listOfAttachments[0], 'attachment-1')
+  it('clicking on Only use messages for non-urgent needs should open largePanel', () => {
+      fireEvent.press(screen.getByLabelText('Only use messages for non-urgent needs'))
+      expect(mockNavigationSpy).toHaveBeenCalled()
     })
-  })
\ No newline at end of file
+
+  it('clicking on attachment should call onPressAttachment(), which calls downloadFileAttachment() from store/actions', () => {
+    fireEvent.press(screen.getByText('testAttachment (1 MB)'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/ViewMessage/MessageCard.test.tsx:91]:
```suggestion
    fireEvent.press(screen.getByRole('button', { name: 'testAttachment (1 MB)' }))
```

File: VAMobile/src/screens/HealthScreen/SecureMessaging/ViewMessage/ViewMessageScreen.test.tsx [hunk 10]
```diff
@@ -146,7 +143,7 @@ context('ViewMessageScreen', () => {
       beforeEach(() => {
         initializeTestInstance(mockMessagesById, mockThreads, false, 3, [1])
       })
-      it('should show AlertBox with "Message could not be found" title', async () => {
+      it('should show AlertBox with "Message could not be found" title', () => {
         expect(screen.getByText('mock sender 1')).toBeTruthy()
         fireEvent.press(screen.getByText('mock sender 1'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/ViewMessage/ViewMessageScreen.test.tsx:148]:
```suggestion
        expect(screen.getByRole('tab', { name: 'mock sender 1' })).toBeTruthy()
        fireEvent.press(screen.getByRole('tab', { name: 'mock sender 1' }))
```

**Review by theodur — CHANGES_REQUESTED** [review]

File: VAMobile/src/screens/HealthScreen/SecureMessaging/CernerAlertSM/CernerAlertSM.test.tsx [hunk 11]
```diff
@@ -72,15 +71,15 @@ context('CernerAlertSM', () => {
...
     it('should show all facility names', () => {
-      fireEvent.press(screen.getByText("Make sure you're in the right health portal"))
+      fireEvent.press(screen.getByRole('tab', { name: "Make sure you're in the right health portal" }))
       expect(screen.getByText("FacilityOne")).toBeTruthy()
       expect(screen.getByText("FacilityTwo")).toBeTruthy()
     })
   })
 
   it('should only show cerner facilities, not other facilities and should call mockExternalLinkSpy when link is selected', () => {
     expect(screen.getByText("Make sure you're in the right health portal")).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/CernerAlertSM/CernerAlertSM.test.tsx:81]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/EditDraft/EditDraft.test.tsx [hunk 12]
```diff
@@ -226,42 +224,42 @@ context('EditDraft', () => {
...
       initializeTestInstance({ loading: true })
       expect(screen.getByText("Loading your draft...")).toBeTruthy()
     })
   })
 
   describe('when there is an error', () => {
-    it('should display the ErrorComponent', async () => {
+    it('should display the ErrorComponent', () => {
       initializeTestInstance({ screenID: ScreenIDTypesConstants.SECURE_MESSAGING_COMPOSE_MESSAGE_SCREEN_ID })
       expect(screen.getByText("The app can't be loaded.")).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/EditDraft/EditDraft.test.tsx:250]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/EditDraft/EditDraft.test.tsx [hunk 13]
```diff
@@ -270,7 +268,7 @@ context('EditDraft', () => {
   })
 
   describe('on click of add files button', () => {
-    it('should call useRouteNavigation', async () => {
+    it('should call useRouteNavigation', () => {
       fireEvent.press(screen.getByLabelText('Add Files'))
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/EditDraft/EditDraft.test.tsx:272]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/EditDraft/EditDraft.test.tsx [hunk 14]
```diff
@@ -226,42 +224,42 @@ context('EditDraft', () => {
...
       fireEvent.press(screen.getByLabelText('Only use messages for non-urgent needs'))
       expect(mockNavigationSpy).toHaveBeenCalled()
     })
   })
 
   describe('when pressing the back button', () => {
-    it('should ask for confirmation if any field filled in', async () => {
+    it('should ask for confirmation if any field filled in', () => {
       fireEvent.changeText(screen.getByTestId('messageText'), 'Random String')
       fireEvent.press(screen.getByText('Cancel'))
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/EditDraft/EditDraft.test.tsx:264]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/NotEnrolledSM/NotEnrolledSM.test.tsx [hunk 15]
```diff
@@ -21,7 +20,7 @@ context('NotEnrolledSM', () => {
   })
 
   describe('when Learn how to upgrade link is clicked', () => {
-    it('should launch external link', async () => {
+    it('should launch external link', () => {
       fireEvent.press(screen.getByTestId('Learn how to upgrade to a My HealtheVet Premium account'))
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/NotEnrolledSM/NotEnrolledSM.test.tsx]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:96]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx (hunk 3 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:107]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx (hunk 4 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:117]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx (hunk 5 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:137]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx (hunk 6 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:158]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx (hunk 7 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:181]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx (hunk 8 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.test.tsx:204]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/ViewMessage/MessageCard.test.tsx (hunk 9 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/ViewMessage/MessageCard.test.tsx:91]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/ViewMessage/ViewMessageScreen.test.tsx (hunk 10 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/ViewMessage/ViewMessageScreen.test.tsx:148]: updated

File: VAMobile/src/screens/HealthScreen/SecureMessaging/CancelConfirmations/ComposeCancelConfirmation.test.tsx (hunk 1 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/SecureMessaging/CancelConfirmations/ComposeCancelConfirmation.test.tsx:3]: removed

File: VAMobile/src/screens/HealthScreen/SecureMessaging/ReplyMessage/ReplyMessage.test.tsx [hunk 16]
```diff
@@ -1,13 +1,11 @@
 import 'react-native'
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/SecureMessaging/ReplyMessage/ReplyMessage.test.tsx:1]: Looks like this one was missed

**Review by theodur — APPROVED** [review]

**Review by alexandec — APPROVED** [review]

**Pull request merged**